src/Compilers/CSharp/Test/Emit3/UnionsTests.cs

```diff
@@ -3460,5 +3472,861 @@ static void Test2(object u)
                 Diagnostic(ErrorCode.ERR_PatternWrongType, "C3").WithArguments("S1", "C3").WithLocation(202, 33)
                 );
         }
+
+        [Fact]
+        public void Exhaustiveness_01()
+        {
+            var src = @"
+struct S1 : System.Runtime.CompilerServices.IUnion
+{
+    private readonly object _value;
+    public S1(int x) { _value = x; }
+#nullable enable
+    public S1(string? x) { _value = x; }
+#nullable disable
+    object System.Runtime.CompilerServices.IUnion.Value => _value;
+}
+
+class Program
+{
+    static int Test1(S1 u)
+    {
+#line 100
+        return u switch { int => 1, string => 2, null => 3 };
+    } 
+
+    static int Test2(S1 u)
+    {
+#line 200
+        return u switch { int => 1, null => 3, string => 2 };
+    } 
+
+    static int Test3(S1 u)
+    {
+#line 300
+        return u switch { null => 3, int => 1, string => 2 };
+    } 
+
+    static int Test4(S1 u)
+    {
+#line 400
+        return u switch { int => 1, string => 2 };
+    }   
+
+    static int Test5(S1 u)
+    {
+#nullable enable
+#line 500
+        return u switch { int => 1, string => 2 };
+#nullable disable
+    }   
+
+    static int Test6(S1 u)
+    {
+#line 600
+        return u switch { int => 1, null => 3 };
+    }   
+
+    static int Test7(S1 u)
+    {
+#line 700
+        return u switch { null => 3, int => 1 };
+    }   
+
+    static int Test8(S1 u)
+    {
+#line 800
+        return u switch { int => 1 };
+    }   
+
+    static int Test9(S1 u)
+    {
+#line 900
+        return u switch { not int => 1 };
+    }   
+
+    static int Test10(S1 u)
+    {
+#line 1000
+        return u switch {  null => 3, not int => 1 };
+    }   
+}
+";
+            var comp = CreateCompilation([src, IUnionSource]);
+            var verifier = CompileAndVerify(comp).VerifyDiagnostics(
+                // (100,50): hidden CS9335: The pattern is redundant.
+                //         return u switch { int => 1, string => 2, null => 3 };
+                Diagnostic(ErrorCode.HDN_RedundantPattern, "null").WithLocation(100, 50),
```

Comment:
It looks like this and several other similar diagnostics are covered by a prototype comment further down. I agree that the behavior is not desirable

Reply:
> It looks like this and several other similar diagnostics are covered by a prototype comment further down. I agree that the behavior is not desirable

No, it is "normal" to get a hidden diagnostic like that. See [https://github.com/dotnet/roslyn/pull/81442#discussion_r2594498683](https://github.com/dotnet/roslyn/pull/81442#discussion_r2594498683)